pom.xml

```diff
@@ -26,6 +26,22 @@
     <build>
         <pluginManagement>
             <plugins>
+                <plugin>
+                    <groupId>io.github.git-commit-id</groupId>
```

Comment:
I'm stuck here

when run mvn build command, I got an error
```
[ERROR] Failed to execute goal io.github.git-commit-id:git-commit-id-maven-plugin:6.0.0:revision (get-the-git-infos) on project testing-base: Git command exited with invalid status [128]: directory: `/home/tungtv/workplace/JAMES/tmail-backend/.git/modules`, command: `git describe --always --dirty=-dirty --match=* --abbrev=7`, stdout: ``, stderr: `fatal: this operation must be run in a work tree`
```
I tried to ignore this plugin, but unsuccessful

Reply:
the simplest way is reverting this plugin version in james-project

Reply:
The bug from plugin: https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/639

Reply:
https://github.com/git-commit-id/git-commit-id-maven-plugin#plugin-compatibility-with-java 
It seems does not officially support Java 17.

What is this plugin anyway? Can we just remove it?

Reply:
I first thought maybe it's related to java 11 vs java 17, as it was green somehow on apache CI but not here... But even when I tried to compile james project with it and java11, I still get the same issue.

Ok to revert it. I don't understand though how it was green on the apache CI? And my guess is you compiled with it before to, must have bee fine as well?

Reply:
No I think it corrupted somehow my local instal... Agree with Quan, try to see if we cant simply remove it. if not, revert it

Reply:
> Why do you add this plugin in the first place?

I just upgrade the version of the plugin, https://github.com/apache/james-project/pull/1657/commits/ba76de8586f018efec8615ad22dfc11efdd8fe86

>  I don't understand though how it was green on the apache CI

It still works in james-project,  mvn builds success, 
In tmail-backend, it fails because "git submodule" and this is a bug of plugin (https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/639)

______
If it has configured as the picture below, it works

![image](https://github.com/linagora/tmail-backend/assets/81145350/2eaf15b0-d7ac-45bc-9042-20e73f8f36a2)

Reply:
> What is this plugin anyway? 

This plugin was added a very long time (since 2018)
When I search history, maybe It is a plugin for the developer to check which latest commit in deployed. 
I'm not sure, I just upgrade the version as the tool suggested.

Reply:
> Can we just remove that plugin in TMail

How? I wish I can do that, as you see in the last commit (https://github.com/linagora/tmail-backend/pull/766/commits/10f69d31207484ba8901ce3e153e21140ea8f994)
I tried to ignore it but was unsuccessful, the parent pom CAN NOT override the child

Reply:
revert pr: https://github.com/apache/james-project/pull/1667